### PR TITLE
Feature/transaction

### PR DESCRIPTION
**Pull request by Anant-Raman**

### CODE REVIEW

File: app/src/main/java/com/example/mobiledger/common/utils/DialogUtils.kt [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package com.example.mobiledger.common.utils
+
+import androidx.fragment.app.FragmentManager
+import com.example.mobiledger.common.extention.getName
+import com.example.mobiledger.presentation.recordtransaction.RecordTransactionDialogFragment
```
Comment (akhilj33) [inline app/src/main/java/com/example/mobiledger/common/utils/DialogUtils.kt:5]: Indentation needs to  be corrected in every class. Set you android studio's hard wrap property to 140 and then indent

File: app/src/main/java/com/example/mobiledger/data/repository/TransactionRepository.kt [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    suspend fun getMonthlyTransactionFromFirebaseDb(uid: String, monthYear: String): AppResult<MonthlyTransactionSummaryEntity?>
+    suspend fun addUserTransactionToFirebase(
+        uid: String,
+        monthYear: String,
+        transactionId: String,
+        transaction: TransactionEntity,
+        monthlyTransactionSummaryEntity: MonthlyTransactionSummaryEntity?
+    ): Boolean
+}
+
```
Comment (akhilj33) [inline app/src/main/java/com/example/mobiledger/data/repository/TransactionRepository.kt:21]: Update Transaction id to time in millis from counters

File: app/src/main/java/com/example/mobiledger/data/sources/api/model/UserApi.kt [hunk 3]
```diff
@@ -17,6 +19,18 @@ const val USERS = "Users"
 const val EMAIL_ID = "emailId"
 const val USER_NAME = "userName"
 const val PHONE_NUMBER = "phoneNo"
+const val MONTH = "Months"
+const val TRANSACTION = "Transaction"
```
Comment (akhilj33) [inline app/src/main/java/com/example/mobiledger/data/sources/api/model/UserApi.kt]: Create New Util Class

File: app/src/main/java/com/example/mobiledger/data/repository/TransactionRepository.kt (hunk 2 above)
Comment (Anant-Raman) [reply app/src/main/java/com/example/mobiledger/data/repository/TransactionRepository.kt:21]: Done

File: app/src/main/java/com/example/mobiledger/data/sources/api/model/UserApi.kt (hunk 3 above)
Comment (Anant-Raman) [reply app/src/main/java/com/example/mobiledger/data/sources/api/model/UserApi.kt]: Done

File: app/src/main/java/com/example/mobiledger/common/utils/DialogUtils.kt (hunk 1 above)
Comment (Anant-Raman) [reply app/src/main/java/com/example/mobiledger/common/utils/DialogUtils.kt:5]: Didn't find a lot of place, but did wherever possible

**Pull request merged**